src/core/opamSystem.ml

```diff
@@ -177,18 +177,29 @@ let write file contents =
 
 let setup_copy ?(chmod = fun x -> x) ~src ~dst () =
   let ic = open_in_bin src in
-  let oc =
-    try
-      let perm =
-        (Unix.fstat (Unix.descr_of_in_channel ic)).st_perm |> chmod
-      in
+  try
+    let perm =
+      (Unix.fstat (Unix.descr_of_in_channel ic)).st_perm |> chmod
+    in
+    let () =
+      try if Unix.((lstat dst).st_kind <> S_REG) then
+            remove_file dst
```

Comment:
This is the right behaviour, but now I am wondering what happens if a package overwrites a directory installed by another package with a file :thinking: 
It shouldn't be worse than just overriding files, and since this `DirTrack` is done outside of this, it should correctly detect the deleted files; but this has consequences with #4494, since that only handles lists of files expected to be installed: it should also track anything *below* in case it encounters a directory right ?

That seems pretty minor, though.

Reply:
I'm not sure what happens at the higher level, but `OpamSystem.install` will fail if the destination is a directory?

Reply:
Several kind of files are written in the install process:
* the `pkg.config` file that is package specific by name
* the files in `.install` file that are written in `prefix/pkg/xxx`
* the misc files in `.install` that need confirmation from user to be installed, as they are absolute paths
Other install/copy are done via `install` field commands not opam instructions

Reply:
> I'm not sure what happens at the higher level, but OpamSystem.install will fail if the destination is a directory?

indeed, it is the first check
```
let install ?(warning=default_install_warning) ?exec src dst =
  if Sys.is_directory src then
    internal_error "Cannot install %s: it is a directory." src;
  if (try Sys.is_directory dst with Sys_error _ -> false) then
    internal_error "Cannot install to %s: it is a directory." dst;
```